packages/rds/src/components/Heading/Heading.tsx

```diff
@@ -0,0 +1,27 @@
+import {ComponentChildren} from 'preact';
+
+import {Text, FontWeight} from '../Text/Text.js';
+
+export type HeadingLevel = 1 | 2 | 3 | 4 | 5 | 6;
+export type HeadingComponent = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
+
+export type HeadingProps = preact.JSX.HTMLAttributes & {
+  level?: HeadingLevel;
+  size?: string;
```

Comment:
If this is just `string` it doesn't give any hints to the possible values. Maybe we create in `Text.tsx` something like the following. The cherry on top is if we can emit (or generate) the possible values somehow for stronger validation/checking.

```
/** 
 * Describe what the possible values are and how to use tokens.
 */
type Size = string;
```